server-adapters/resteasy-undertow-spring/src/test/java/org/jboss/resteasy/springmvc/test/spring/AsynchSpringTest.java

```diff
@@ -0,0 +1,209 @@
+package org.jboss.resteasy.springmvc.test.spring;
+
+import io.undertow.servlet.api.DeploymentInfo;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.jboss.resteasy.core.AsynchronousDispatcher;
+import org.jboss.resteasy.plugins.server.undertow.spring.UndertowJaxrsSpringServer;
+import org.jboss.resteasy.springmvc.test.client.BasicSpringTest;
+import org.jboss.resteasy.test.TestPortProvider;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.web.servlet.DispatcherServlet;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation.Builder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class AsynchSpringTest {
+   private static CountDownLatch latch;
+
+   UndertowJaxrsSpringServer server;
+
+   @Before
+   public void before() {
+      server = new UndertowJaxrsSpringServer();
+
+      server.start();
+
+      DeploymentInfo deployment = server.undertowDeployment("classpath:spring-test-async-server.xml", null);
+      deployment.setDeploymentName(BasicSpringTest.class.getName());
+      deployment.setContextPath("/");
+      deployment.setClassLoader(BasicSpringTest.class.getClassLoader());
+
+      server.deploy(deployment);
+   }
+
+   @After
+   public void after() {
+      server.stop();
+   }
+
+   @Test
+   public void testOneway() throws Exception {
+      latch = new CountDownLatch(1);
+      Client client = ResteasyClientBuilder.newClient();
```

Comment:
As above, the client needs to be closed

Reply:
I have put the client open and close code into `@Before` and `@After` methods. So the following points are fixed.